integrations/langflow/src/stepflow_langflow_integration/cli/main.py

```diff
@@ -0,0 +1,405 @@
+def convert(
+    input_file: Path, 
+    output_file: Optional[Path],
+    pretty: bool,
+    validate: bool
+):
+    """Convert a Langflow JSON workflow to Stepflow YAML.
+    
+    If no output file is specified, prints the YAML to stdout.
+    """
+    try:
+        converter = LangflowConverter(validate_schemas=validate)
+        stepflow_yaml = converter.convert_file(input_file)
+        
+        if output_file:
+            # Write to file
+            with open(output_file, "w", encoding="utf-8") as f:
+                f.write(stepflow_yaml)
+            click.echo(f"✅ Successfully converted {input_file} to {output_file}", err=True)
+        else:
+            # Write to stdout
+            click.echo(stepflow_yaml)
+        
+    except (ConversionError, ValidationError) as e:
+        click.echo(f"❌ Conversion failed: {e}", err=True)
+        sys.exit(1)
+    except Exception as e:
+        click.echo(f"❌ Unexpected error: {e}", err=True)
+        sys.exit(1)
+
+
+@main.command()
+@click.argument("input_file", type=click.Path(exists=True, path_type=Path))
+def analyze(input_file: Path):
+    """Analyze a Langflow workflow structure.
+    
+    This command provides detailed analysis of a Langflow JSON workflow without converting it.
+    It examines the workflow structure, component types, dependencies, and identifies potential
+    issues that might affect conversion. Useful for understanding workflow complexity and 
+    debugging conversion problems before attempting the full conversion process.
+    """
+    try:
+        converter = LangflowConverter()
+        
+        # Load and analyze
+        with open(input_file, "r", encoding="utf-8") as f:
+            langflow_data = json.load(f)
+        
+        analysis = converter.analyze(langflow_data)
+        
+        # Display results
+        click.echo(f"📊 Analysis of {input_file}:")
+        click.echo(f"  • Nodes: {analysis['node_count']}")
+        click.echo(f"  • Edges: {analysis['edge_count']}")
+        
+        click.echo("\n📦 Component Types:")
+        for comp_type, count in analysis["component_types"].items():
+            click.echo(f"  • {comp_type}: {count}")
+        
+        if analysis["dependencies"]:
+            click.echo("\n🔗 Dependencies:")
+            for target, sources in analysis["dependencies"].items():
+                click.echo(f"  • {target} ← {', '.join(sources)}")
+        
+        if analysis["potential_issues"]:
+            click.echo("\n⚠️  Potential Issues:")
+            for issue in analysis["potential_issues"]:
+                click.echo(f"  • {issue}")
+        
+    except (ConversionError, ValidationError) as e:
+        click.echo(f"❌ Analysis failed: {e}", err=True)
+        sys.exit(1)
+    except Exception as e:
+        click.echo(f"❌ Unexpected error: {e}", err=True)
+        sys.exit(1)
+
+
+@main.command()
+@click.option("--host", default="localhost", help="Server host")
```

Comment:
Should add a more explicit option for "stdio" vs. "http". Note that we don't yet support http.